src/battle-dex.ts

```diff
@@ -193,12 +193,12 @@ const Tools = {
 
 	resourcePrefix: (() => {
 		let prefix = '';
-		if (document.location!.protocol !== 'http:') prefix = 'https:';
+		if (!document.location || document.location.protocol !== 'http:') prefix = 'https:';
```

Comment:
@Zarel 

Wasn't sure what to do here. An alternate solution is to leave this alone and stub `document.location` in the test file.

Reply:
This is good.

Reply:
I'm surprised `document` exists at all here. Are you using Node or...?